### PR TITLE
Simplify extract types to support RT v2 pipeline

### PR DESCRIPTION
**Pull request by atvaccaro**
This PR has been published as version 2022.8.23.

The primary change in this PR is splitting up the GTFS extract types. In general I'm trying to move to simpler Pydantic types and use functions to implement additional behavior as necessary.

### CODE REVIEW

**Review by lauriemerrell — APPROVED** [review]
Just one inline not-really-a-question-more-of-a-comment. Also -- are you waiting to merge https://github.com/cal-itp/data-infra/pull/1691 until this merges? If so, I like that. I don't love referencing unmerged `calitp` code in merged `data-infra` code.

TLDR: Ok to merge, thanks for breaking these back out.

File: calitp/storage.py [hunk 1]
```diff
@@ -508,12 +505,11 @@ def get_latest(cls) -> "AirtableGTFSDataExtract":
         )
 
 
-class GTFSFeedExtractInfo(PartitionedGCSArtifact):
-    partition_names: ClassVar[List[str]] = ["dt", "base64_url", "ts"]
+class GTFSFeedExtract(PartitionedGCSArtifact, ABC):
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply calitp/storage.py:508]: 1. Yeah I think we were trying to figure out how to distinguish between containers and data/records via names. In this case, I think the "extract" is relatively obvious to mean the container? I've started to use `Record` as the suffix for actual data where we have a type for it.
2. Yeah sorry I think we'll get better stability over time

**Comment by atvaccaro** [comment]
> Just one inline not-really-a-question-more-of-a-comment. Also -- are you waiting to merge [cal-itp/data-infra#1691](https://github.com/cal-itp/data-infra/pull/1691) until this merges? If so, I like that. I don't love referencing unmerged `calitp` code in merged `data-infra` code.
> 
> TLDR: Ok to merge, thanks for breaking these back out.

Yep! I did want to wait.
